### PR TITLE
VYGR-349 total removal of secretenvvar

### PR DESCRIPTION
**Pull request by ychen-atlassian**
This PR totally removes the `secretenvvar` and `podsecretenvvar` plugins. It does so by using the new `vars` returned by the `BindableEnvironmentVariables` shape to allow autowiring to construct a direct list of references in the secret.

### How does this work

**Previous Behavior:** secretenvvar plugin scans through *all* dependencies passed to it. It spits out everything as an environment variable
**New Behavior:** the producer decides the environment variable name (a bit weird, yes) and prefix. The consumer sets up dependencies and loops through environment variable names, writing them into the spec of the secret plugin. Secret plugin handles some simple secret data json conversion.

### How to review this PR

I.e. holy shit it's so big. In reality there's only a 460 line code change, the rest is killing the secretenvvar plugin and rewriting all the bundle test yamls.

I've split this PR into multiple commits. Most of the changes are tests. I've moved that into a different commit. You could be able to review each commit independently.

1. Adding testing for the new plugin behavior. This should give an overview of what the new bundle looks like. (+1600 lines inside yaml files)
2. Returning the correct shape for all resources. This actually converts the resources (+300 lines)
3. Removal of the secretenvvar plugin and bug fixes to allow ec2compute and kubecompute to use the new shapes (-1500 lines)
4. More bug fixes.

### Example Diff

image
image

You'll notice there are some missing environment variables now that we've done this.

- CREATION_TIMESTAMP - internal
- IAMPOLICYSNIPPET - internal
- TEMPLATE_NAME - internal

### CODE REVIEW

File: pkg/orchestration/wiring/asapkey/plugin.go [hunk 1]
```diff
@@ -43,11 +46,22 @@ func New() *WiringPlugin {
...
 }
 
+func shapes(resource *orch_v1.StateResource, smithResource *smith_v1.Resource, context *wiringplugin.WiringContext) ([]wiringplugin.Shape, error) {
+	bindableEnvVarShape := knownshapes.NewBindableEnvironmentVariables(smithResource.Name, ResourcePrefix, map[string]string{
+		"PRIVATE_KEY": "data.private_key",
+		"ISSUER":      "data.issuer",
+		"KEY_ID":      "data.key_id",
+		"AUDIENCE":    "data.audience",
+	})
+	bindableEnvVarShape.Data.ExcludeResourceNameInKey = true
```
Comment (ash2k) [inline pkg/orchestration/wiring/asapkey/plugin.go]: Why not make this field a constructor parameter?

File: pkg/orchestration/wiring/aws/plugin.go [hunk 2]
```diff
@@ -126,13 +123,6 @@ func instanceSpec(serviceName voyager.ServiceName, resourceName string, oapName
...
-func objectMeta(prefix oap.EnvVarPrefix) meta_v1.ObjectMeta {
-	annotations := make(map[string]string)
-	if prefix != "" {
-		// This is needed for compatibility with former Micros RPS resource types
-		// that generate environment variables with custom prefix
-		// e.g. DYNAMO_* instead of DYNAMO_DB_* for dynamo-db
-		annotations[voyager.Domain+"/envResourcePrefix"] = string(prefix)
-	}
-	return meta_v1.ObjectMeta{Annotations: annotations}
+func objectMeta() meta_v1.ObjectMeta {
```
Comment (ash2k) [inline pkg/orchestration/wiring/aws/plugin.go]: Kill this function

File: pkg/orchestration/wiring/aws/resources.go [hunk 3]
```diff
@@ -7,49 +7,140 @@ import (
...
+		//     internal outputs above the orchestration layer.
+		//  4. Structure is defined in the output spec of autowiring (k8s vs
+		//     ec2) as opposed to having bespoke logic in the plugin.
+		//
+		// The secretenvvar approach (originally) needed to:
+		//  - pull down all secrets for all resources with the env shape
+		//  - run either podsecretenvvar or secretenvvar also passing through a
+		//    renameMap and ignoreKeyRegex, and just dump everything into the
+		//    environment variables that don't match ignoreKeyRegex.
+		return nil, errors.Errorf("cfn template %q is not supported", templateName)
```
Comment (ash2k) [inline pkg/orchestration/wiring/aws/resources.go]: `CloudFormation template ...`

File: pkg/orchestration/wiring/rds/plugin.go [hunk 4]
```diff
@@ -82,11 +89,31 @@ func New() *WiringPlugin {
 			InstanceSpec:                  instanceSpec,
 			ObjectMeta:                    objectMeta,
 			ResourceType:                  ResourceType,
-			OptionalShapes:                svccatentangler.NoOptionalShapes,
+
+			OptionalShapes: shapes,
```
Comment (ash2k) [inline pkg/orchestration/wiring/rds/plugin.go]: nit: Don't need empty line above. In other places too.

File: pkg/orchestration/wiring/wiringutil/knownshapes/bindable_environment_variables.go [hunk 5]
```diff
@@ -21,6 +21,7 @@ type BindableEnvironmentVariablesData struct {
 	wiringplugin.BindableShapeStruct `json:",inline"`
 	Prefix                           string            `json:"prefix,omitempty"`
 	Vars                             map[string]string `json:"vars,omitempty"`
+	ExcludeResourceNameInKey         bool              `json:"exclude_resource_name_in_key,omitempty"`
```
Comment (ash2k) [inline pkg/orchestration/wiring/wiringutil/knownshapes/bindable_environment_variables.go]: camelCase please

File: pkg/execution/plugins/generic/secretplugin/BUILD.bazel [hunk 6]
```diff
@@ -17,3 +17,16 @@ go_library(
         "//vendor/k8s.io/apimachinery/pkg/runtime:go_default_library",
     ],
 )
+
+go_test(
```
Comment (ash2k) [inline pkg/execution/plugins/generic/secretplugin/BUILD.bazel]: size is missing

File: pkg/orchestration/wiring/asapkey/plugin.go (hunk 1 above)
Comment (ychen-atlassian) [reply pkg/orchestration/wiring/asapkey/plugin.go]: The field is usually false, it seemed like a bit tedious to always provide false as an argument.

I don't have a strong opinion here so I can just add it into the constructor.

**Comment by wryun** [comment]
@ychen-atlassian +++ and @ash2k !!!

Sweet! ... also, I can see this? Whoah! We should meet up for a chat.

**Comment by ash2k** [comment]
@wryun you know where to find us =) come any day! Or we can meet somewhere else if you prefer.

**Pull request merged**